docs/forwarder/README.md

```diff
@@ -0,0 +1,163 @@
+# Forwarder
+
+## Overview
+
+The forwarder verifies that the a report comes from a valid oracle network and routes it to a receiver program.
+
+Let's look at the accounts associated with a report instruction to understand what it does.
+
+```
+#[derive(Accounts)]
+#[instruction(data: Vec<u8>)]
+pub struct Report<'info> {
+    pub state: Account<'info, ForwarderState>,
+
+    #[account(
+        mut,
+        seeds = [b"config", state.key().as_ref(), &extract_config_id(extract_raw_report(&data))],
+        bump
+    )]
+    oracles_config: Account<'info, OraclesConfig>,
+
+    #[account(mut)]
+    pub transmitter: Signer<'info>,
+
+    /// CHECK: This is a PDA
+    #[account(seeds = [b"forwarder", state.key().as_ref()], bump = state.authority_nonce)]
+    pub forwarder_authority: UncheckedAccount<'info>,
+
+    // it is dependent on the state.key(), a predetermined bump, workflow execution id, config_id, report_id
+    #[account(
+        init_if_needed,
+        payer = transmitter,
+        space = ANCHOR_DISCRIMINATOR + ExecutionState::INIT_SPACE,
+        seeds = [
+            b"execution_state", 
+            state.key().as_ref(),
+            &extract_transmission_id(extract_raw_report(&data), receiver_program.key)
+        ],
+        bump
+    )]
+    pub execution_state: Account<'info, ExecutionState>,
+
+    #[account(executable)]
+    /// CHECK: We don't use Program<> here since it can be any program, "executable" is enough
+    pub receiver_program: UncheckedAccount<'info>,
+
+    pub system_program: Program<'info, System>,
+    // remaining accounts passed to receiver
+}
+```
+
+`state` is a keypair, program owned account which represents the forwarder instance we are working with. Think of this is a specific forwarder instance.
+
+`oracles_config` is a PDA which stores DON info
+
+`forwarder_authority` is the PDA which is responsible for signing the CPI to a receiver program via invoke_signed()
+
+`execution_state` is the PDA which stores a report's execution status. For now, it only stores successes.
```

Comment:
how will offchain components (writetarget) know when a transmission is failed / fatal?

Reply:
Good question. This is something that the platform expansion team will need to add in later iterations. I think the body of work is too large for our timeline at the moment. But I have some thoughts [here](https://chainlink-core.slack.com/archives/C024M6N5MS8/p1743181059358989). I think the first option is the most robust, but we have to have a new capabilities for that to happen.

As it is right now, the write targeter will just keep on trying until it reaches the max tries and then stop.